Core/ECS/Entity.hpp

```diff
@@ -0,0 +1,33 @@
+#include <ECS/Utils/IDUtils.hpp>
+#include <ECS/Managers/ComponentManager.hpp>
+
+template<typename T>
+class Entity {
+
+    // unique id for entity
+    static const ObjectID ENTITY_ID;

```

Comment:
I dont think this should be static const, as diffrent entities would be assigned different UUIDs

Reply:
Maybe i misunderstood, is this Entity_ID for multiple instances, i.e <GameObject> or <Gun>, or is this an ID that should be unique to each instance?

Reply:
But if it's static const, each instance of the entity will have the same ENTITY_ID.